src/systolic_array/unit_test/test_tensor_iterator.cpp

```diff
@@ -0,0 +1,149 @@
+#include <iostream>
+#include <vector>
+#include <memory>
+
+#include "aladdin/unit-test/catch.hpp"
+#include "systolic_array/tensor.h"
+
+using namespace systolic;
+
+// Compute the reference results for region-based iterator operation. It's slow
+// as it literally converts the region into single step movements, but it gives
+// us the correct results.
+std::vector<int> computeReferenceResults(const TensorIndexIterator& iter,
+                                         const std::vector<int>& region) {
+  std::unique_ptr<TensorIndexIterator> refIter(iter.clone());
```

Comment:
with a copy ctor this would be as simple as `TensorIndexIterator refIter(iter)`.

Reply:
See the first comment.